File: sassfnb/src/main/java/com/sassfnb/application/service/OrderWorkflowService.java
package com.sassfnb.application.service;

import com.sassfnb.adapters.rest.dto.order.OrderDtos.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface OrderWorkflowService {

    // ===== Staff/POS (auth + tenant context) =====
    SubmitOrderResponse submit(UUID orderId);

    ReopenOrderResponse reopen(UUID orderId, ReopenOrderRequest req);

    Page<OrderResponse> staffList(StaffOrderListRequest req, Pageable pageable);

    OrderResponse staffGet(UUID orderId);

    OrderResponse fire(UUID orderId, FireOrderRequest req);

    OrderResponse voidItem(UUID orderId, UUID itemId, VoidItemRequest req);

    OrderResponse voidOrder(UUID orderId, VoidOrderRequest req);

    OrderResponse patchNote(UUID orderId, PatchOrderNoteRequest req);

    // ===== Public (NO tenant header/token) =====
    SubmitOrderResponse submitPublic(UUID orderId, UUID outletId);

    ReopenOrderResponse reopenPublic(UUID orderId, UUID outletId, ReopenOrderRequest req);

    // ===== Part 7 =====
    OrderTotalsResponse totals(UUID orderId);

    CloseOrderResponse close(UUID orderId, CloseOrderRequest req);
}

File: sassfnb-fe/src/api/services/orderWorkflow.service.ts
import { http } from "../client";

export const orderWorkflowService = {
  totals: (orderId: string) => http.get(`/orders/${orderId}/totals`),

  finalize: (orderId: string, payload?: { note?: string | null }) =>
    http.post(`/orders/${orderId}/finalize`, payload ?? null),

  submit: (orderId: string) => http.post(`/orders/${orderId}/submit`, null),

  reopen: (orderId: string, payload?: { reason?: string | null }) =>
    http.post(`/orders/${orderId}/reopen`, payload ?? null),

  fire: (orderId: string, payload: { all?: boolean; itemIds?: string[] }) =>
    http.post(`/orders/${orderId}/fire`, payload),

  voidItem: (orderId: string, itemId: string, payload?: { reason?: string | null }) =>
    http.post(`/orders/${orderId}/items/${itemId}/void`, payload ?? null),

  voidOrder: (orderId: string, payload?: { reason?: string | null }) =>
    http.post(`/orders/${orderId}/void`, payload ?? null),

  patchNote: (orderId: string, payload: { note?: string | null }) =>
    http.patch(`/orders/${orderId}/note`, payload),
};
